### DISCUSSION TITLE
'noindex' is added in my production deployment

### DISCUSSION THREAD

**Discussion by Nyleor** [vercel/community#5741] [Help] [answered]
### Summary

Hello,

I've discovered the following code on my website:

`<meta name="robots" content="noindex"/>`

Upon researching, I learned that this tag is automatically applied to preview deployments, yet the deployment in question is in production.

It is mentioned here (https://vercel.com/docs/deployments/preview-deployments) :

« When a custom domain is assigned to a Git branch, this behavior is prevented. In that case, the X-Robots-Tag header is not set. »

However, this is precisely my situation. My domain is linked to my 'main' Git branch, and in the domain list, my domain is tagged as "Production". In the deployment list, the "current" is indeed on a production build.

I have tried in vain to override this setting in the headers in next.config.js, by creating a vercel.json, or directly in the metadata in my layout, but nothing changes but this : when I do `curl -I mydomain.com`, I get « X-Robots-Tag: index, follow ». But I still have the 'noindex' tag in the source code.

Thank's for your help !

### Steps to Reproduce

Just pushing my changes to my Git main branch, and let Vercel building it into production. And then, when I visit my website, I find the noindex tag in the source code.

**Comment by yurickv** [comment]
i have a  SIMILAR problem, my production deployment includes meta teg  name="robots" content="noindex"/.   Next.js project  on Versel deployment.
i have already force added 
export const metadata: Metadata = {
   robots: {    index: true   },       
};                             
on every page, but a negative value "noindex" overrides "index" (these tags are both in the page source code).

Also have already force added {
  "headers": [
    {
      "source": "/(.*)",
      "headers": [
        {
          "key": "X-Robots-Tag",
          "value": "index, follow"
        }
      ]
    }
  ]
}
 in versel.json  file

the same:  on every page, but a negative value "noindex" overrides "index" (these tags are both in the page source code).

my repo   https://github.com/yurickv/adrenalin-next

the same: Steps to Reproduce
Just pushing my changes to my Git main branch, and let Vercel building it into production. And then, when I visit my website, I find the noindex tag in the source code.

**Comment by yurickv** [comment]
here they discuss the possible cause of the problem https://github.com/vercel/next.js/issues/58615   
the following solution to the problem is proposed https://github.com/vercel/next.js/issues/58615#issuecomment-1820948414
This solution has not worked correctly for me so far, I need to understand it further

But I just commented out the code related to useSearchParams (in my project it's not critical) and the problem with automatically adding the noindex tag **disappeared**!

**Reply by Nyleor** [reply]
Thank you very much for thoses links. I found them before I posted my topic, but there is no useSearchParams in my project, so I need to investigate more.

Anyway, thank you very much for the follow up.

**Comment by Nyleor** [comment, marked as answer]
I finally had the time to look into it more closely. I discovered that the problem, contrary to what I initially thought, did not come from Vercel, but indeed from NextJS. And the "problem" indeed comes from useSearchParams(), which I thought was not used in my project but was in fact utilized by one of the packages I had installed.

I believe I saw that Next JS 14.1 (I was on 14.0) fixes the issue by requiring a <Suspense>, though I haven't explored further.

Thank you.
